kotlin/viewregistry-android/src/main/java/com/squareup/viewregistry/ModalViewContainer.kt

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright 2019 Square Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.squareup.viewregistry
+
+import android.app.Dialog
+import android.content.Context
+import android.support.annotation.StyleRes
+import android.util.AttributeSet
+import android.view.View
+import android.view.ViewGroup
+import android.view.ViewGroup.LayoutParams.WRAP_CONTENT
+import io.reactivex.Observable
+import kotlin.reflect.jvm.jvmName
+
+/**
+ * Class returned by [ModalContainer.forContainerScreen], qv for details.
+ */
+@PublishedApi
+internal class ModalViewContainer
+@JvmOverloads constructor(
+  context: Context,
+  attributeSet: AttributeSet? = null,
+  @StyleRes private val dialogThemeResId: Int = 0,
+  private val modalDecorator: (View) -> View = { it }
+) : ModalContainer<Any>(context, attributeSet) {
+
+  override fun Any.matches(nextModal: Any) = this::class.java == nextModal::class.java
+
+  override fun showDialog(
+    modalScreen: Any,
+    screens: Observable<out Any>,
+    viewRegistry: ViewRegistry
+  ): Dialog {
+    val binding = viewRegistry.getBinding<Any>(modalScreen::class.jvmName)
+    val view = binding.buildView(screens, viewRegistry, this)
+
+    return Dialog(context, dialogThemeResId).apply {
+      setCancelable(false)
```

Comment:
CRUD! Still hard coded.

Reply:
Maybe that's okay. Solving it would be complicated, and making your own subclass is _so_ easy. Should tweak the docs suggesting it's unusual, though.